### PR TITLE
chg: usr: #307: add configuration option for server socket timeout

### PR DESCRIPTION
**Pull request by jordanw-bq**
* Tested & verified working on Qubole cluster, where we were experiencing this issue

### CODE REVIEW

File: rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java [hunk 1]
```diff
@@ -90,6 +91,7 @@
   // default values
   private static final int DEFAULT_BLOCK_SIZE = 1 * 1024 * 1024; // 1MB
   private static final int DEFAULT_SERVER_CONNECT_TIMEOUT = 1000; // ms
+  private static final int DEFAULT_SERVER_SOCKET_TIMEOUT = 3000; // ms
```
Comment (abhishekdas99) [inline rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java]: I would suggest to make the default value to 6 sec as some of the initial S3 calls are taking 2-3 sec. So to be in the safe side, lets make it 6sec

**Comment by abhishekdas99** [comment]
@jordanw-bq Is it possible to add couple of  simple unit tests covering the cases where you are getting a timeout and not getting a timeout exception.

File: rubix-spi/src/test/java/com/qubole/rubix/spi/TestBookKeeperFactory.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    final int aliveCallDelay = 1000;
+    final int socketTimeout = 500;
+    final int connectTimeout = 500;
+
+    startMockServer(true, NO_DELAY, aliveCallDelay);
+
+    final RetryingBookkeeperClient client = createTestBookKeeperClient(socketTimeout, connectTimeout);
+    client.isBookKeeperAlive(); // should throw expected exception due to socket timeout
+  }
+
```
Comment (abhishekdas99) [inline rubix-spi/src/test/java/com/qubole/rubix/spi/TestBookKeeperFactory.java]: Can we add a test where we are able to connect although there is a delay in the server ( delay is less than the socket time out). In that case we are not going to get any exception

Comment (jordanw-bq) [reply rubix-spi/src/test/java/com/qubole/rubix/spi/TestBookKeeperFactory.java]: @abhishekdas99 Added tests to cover more cases

**Pull request merged**